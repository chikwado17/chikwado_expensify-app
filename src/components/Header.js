import React, { Component } from 'react';
import { Link} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';


export const Header = (props) => {
    return ( 
       <div>
            <Link to="/dashboard">
                <h1>Expensify</h1>
            </Link>
           
            <button onClick={props.startLogout}>Logout</button>
       </div>
     );
}

const mapDispatchToProps = (dispatch) => ({

  startLogout: () => dispatch(startLogout())
});


export default connect(undefined,mapDispatchToProps)(Header);