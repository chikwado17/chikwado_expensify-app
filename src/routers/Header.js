import React, { Component } from 'react';
import { NavLink} from 'react-router-dom';




const Header = () => {
    return ( 
       <div>
            <h1>Expensify</h1>
            <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink><br/>
            <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink><br/>
            <NavLink to="/help" activeClassName="is-active">Help Page</NavLink><br/>
       </div>
     );
}

export default Header;