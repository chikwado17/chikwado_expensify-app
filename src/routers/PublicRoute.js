import React from 'react';
import { connect } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';

//destructuring the props from AppRouter->PrivateRouter, getting the properties like path, exact etc using ...rest
export const PublicRoute = ({
    isAuthenticated,
    component:Component,
    ...rest
    }) => ( 
        //checking if user is authenticated, then show the private page else then redirect if not, to a public page.
            <Route {...rest} component={(props) => (
                isAuthenticated ? (
                    <Redirect to="/dashboard"/>
                ) : (
                    <Component {...props} />
                )
        )}/>
     );

 

const mapStateToProps = (state) => ({
    //auth coming from store ->
    isAuthenticated: !!state.auth.uid
});


export default connect(mapStateToProps)(PublicRoute);