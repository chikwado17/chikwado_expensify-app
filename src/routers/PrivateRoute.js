import React from 'react';
import { connect } from 'react-redux';
import { Route, Redirect } from 'react-router-dom';
import Header from '../components/Header';

//destructuring the props from AppRouter->PrivateRouter, getting the properties like path, exact etc using ...rest
export const PrivateRoute = ({
    isAuthenticated,
    component:Component,
    ...rest
    }) => ( 
        //checking if user is authenticated, then show the private page else then redirect if not, to a public page.
            <Route {...rest} component={(props) => (
                isAuthenticated ? (
                    <div>
                        <Header />
                        <Component {...props} />
                    </div>
                    
                ) : (
                    
                  <Redirect to="/"/>
                )
        )}/>
     );

 

const mapStateToProps = (state) => ({
    //auth coming from store ->
    isAuthenticated: !!state.auth.uid
});


export default connect(mapStateToProps)(PrivateRoute);