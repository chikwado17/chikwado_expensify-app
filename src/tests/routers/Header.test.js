import React  from 'react';
import { Header } from '../../routers/Header';
import { shallow } from 'enzyme';


test("should render Header Component correctly", () => {
//rendering component with enzyme shallow API
    const wrapper = shallow(<Header startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
})
