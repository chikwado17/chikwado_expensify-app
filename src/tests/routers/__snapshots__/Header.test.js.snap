// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header Component correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/create"
  >
    Create Expense
  </NavLink>
  <br />
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
    Help Page
  </NavLink>
  <br />
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
